distributed/worker.py

```diff
@@ -3144,18 +3145,25 @@ def parse_memory_limit(memory_limit, nthreads, total_cores=CPU_COUNT):
 
 job_counter = [0]
 
-import functools
 
+cache_loads = LRU(maxsize=100)
 
-@functools.lru_cache(100)
-def cached_function_deserialization(func):
-    return pickle.loads(func)
+
+def loads_function(bytes_object):
+    """ Load a function from bytes, cache bytes """
+    try:
+        result = cache_loads[bytes_object]
+    except KeyError:
+        result = pickle.loads(bytes_object)
+        if len(bytes_object) < 100000:
```

Comment:
Should we move this check to before the cache check? Since large keys will never be stored, the cache check will always fail.

```python
def loads_function(bytes_object):
    if bytes_object < 100000:
        try:
            result = cache_loads[bytes_object]
        except KeyError:
            cache_loads[bytes_object] = result = pickle.loads(bytes_object)
        return result
    return pickle.loads(bytes_object)
```

Reply:
Yeah, good point